File: Data Structures & Algorithms/house-robber-ii/submission-0.cs
public class Solution {
    public int Rob(int[] nums) {
        var s = nums.Length;

        if (s == 1) return nums[0];

        var rob1 = 0;
        var rob2 = 0;

        for(var i = 0; i < s - 1; i ++)
        {
            var n = nums[i];
            var temp = Math.Max(rob1 + n, rob2);
            rob1 = rob2;
            rob2 = temp;
        }

        var rob3 = 0;
        var rob4 = 0;
        for(var i = 1; i < s; i ++)
        {
            var n = nums[i];
            var temp = Math.Max(rob3 + n, rob4);
            rob3 = rob4;
            rob4 = temp;
        }

        return Math.Max(rob2, rob4);

    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        s = 0
        f = 1

        while s != len(nums):
            if nums[s] == nums[f]:
                return nums[s]
            
            f += 1
            if f >= len(nums):
                s += 1
                f = s + 1

